docs/src/main/asciidoc/getting-started-dev-services.adoc

```diff
@@ -0,0 +1,326 @@
+about the expected output.
+You should see in your terminal that the tests are now passing.
+
+== Adding persistence
+
+=== Creating a Panache Entity
+
+1. To add the persistence libraries, run
+----
+quarkus ext add hibernate-orm-panache jdbc-postgresql
+----
+
+The application will record the names of people it greets. Define a simple Hibernate Entity
+by creating a `Greeting.java` class. Add the following content:
+
+----
+@Entity
+public class Greeting extends PanacheEntity {
+String name;
+}
+----
+
+The entity makes use of xref:{doc-guides}hibernate-orm-panache.adoc[Panache], a layer on top of Hibernate.
+ Extending `PanacheEntity` brings in a range of methods for reading, writing, and finding data.
+ Because all the data access methods are on the `Greeting` entity, rather than on a separate data access class,
+ this is an example of the active record pattern.
+
+The `Greeting` table will have one column, a field called `name`.
+
+=== Writing data
+
+To use the new entity, update the `hello` method to start writing some data.
+
+Change the method to the following:
+
+----
+@GET
+@Transactional
+@Produces(MediaType.TEXT_PLAIN)
+public String hello(@QueryParam("name") String name) {
+   Greeting greeting = new Greeting();
+   greeting.name = name;
+   greeting.persist();
+   return "Hello " + name;
+}
+----
+
+Don't forget the `@Transactional` method, which ensures writes are wrapped
+in a transaction.
+
+Generally, you shouldn't do state updates on a `GET` REST method, but here it makes
+trying things out simpler. Let's assume what's being written is an logging "side effect",
+rather than a meaningful state changes!
+
+Try out the updated endpoint by visiting http://localhost:8080/hello?name=Bloom.
+You should see a "Hello Bloom" message.
+
+=== Reading data
+Although the new persistence code seems to be working without errors, how
+do you know anything is being written to the database?
+
+Add a second REST method to `GreetingResource`.
+
+----
+@GET
+@Path("names")
+@Produces(MediaType.TEXT_PLAIN)
+public String names() {
+List<Greeting> greetings = Greeting.listAll();
+   String names = greetings.stream().map(g-> g.name)
+   .collect(Collectors.joining (", "));
+   return "I've said hello to " + names;
+}
+----
+
+To try it out, visit http://localhost:8080/hello?name=Bloom, and then http://localhost/hello/names.
+
+You should see the following message: "I've said hello to Bloom".
+
+== Dev services
+
+Reading and writing to the database seems to be working well, but that's a bit unexpected.
+Where did a PostgresQL database come from? You didn't set anything up.
+
+The database is being managed using xref:{docfile}/dev-services.adoc[dev services].
+Dev services take care of stopping and starting services needed by your application.
+Because you
+included the `postgresql-jdbc` dependency, the database is a containerised PostGresQL database.
+If you'd added `postgresql-mysql` insead, you would have gotten a containerised MySQL database.
+
```

Comment:
I would add the next paragraph as an admonition, also explaining that you need a container runtime.

Reply:
Actually, a reader that does not have a container runtime will fail abruptly. I know that docker is a listed requirement, but an explicit mention might be useful.

We had users (even co-workers) that complain that nothing was working because they didn't have docker or podman.

Reply:
Admonition added, slightly higher up.